src/util/XDRCereal.h

```diff
@@ -124,7 +124,7 @@ xdr_to_string(const T& t, std::string const& name, bool compact = false)
         cereal::JSONOutputArchive ar(
             os, compact ? cereal::JSONOutputArchive::Options::NoIndent()
                         : cereal::JSONOutputArchive::Options::Default());
-        ar(cereal::make_nvp(name, t));
+        xdr::archive(ar, t, name.c_str());
```

Comment:
I am pretty sure it's okay, but I just want to confirm that this does the right thing if `name.empty()` right?

Reply:
Thanks for pointing that out! I didn't think about that possibility. I checked the code and it just prints something like
```
"": {
 ...
}
```
which I doubt will be very useful. I decided to require that the input string be non-empty.

Initially, I felt that it might be an over-kill, but the original issue was that we have the word `value0`, so replacing it with the empty string will not really be much of an improvement, so I figured that I'd require it. Let me know what you think!

Reply:
I think this is ok, but you might want to do a ReleaseAssertOrThrow (to ensure the assert remains inside release builds) and of course make sure tests all pass. maybe also run them at trace log-level in case we missed any cases.

Reply:
@graydon 
- I didn't know about `ReleaseAssert`! Although after some thinking, I started to wonder that if it'd be better if the `assert` does NOT remain inside release builds. I'm currently thinking that this issue of printing `"": {` may not be important enough to warrant an abort or throw in production.
- I just confirmed that all the tests do pass.

Let me know what you think!

Reply:
After more thoughts & learning about https://github.com/stellar/stellar-core/issues/2852, I started thinking that it may be better to not have any assertion statements. Printing `"": {` doesn't seem bad enough to warrant an abort or throw in production.